pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -127,6 +127,10 @@
     AuthenticationDataSource authenticationData;
     AuthenticationProvider authenticationProvider;
     AuthenticationState authState;
+    // In case of proxy, if the authentication credentials are forwardable,
+    // it will hold the credentials of the original client
+    AuthenticationState originalAuthState;
+    private boolean pendingAuthChallengeResponse = false;
```

Comment:
I think this needs to be marked volatile?

Reply:
This variable will be always accessed only the connection IO thread so there won't be any race condition.